packages/dds/tree/src/simple-tree/api/schemaFactory.ts

```diff
@@ -294,18 +295,21 @@ export class SchemaFactory<
 	public object<
 		const Name extends TName,
 		const T extends RestrictiveStringRecord<ImplicitFieldSchema>,
+		const TCustomMetadata = unknown,
 	>(
 		name: Name,
 		fields: T,
+		props?: NodeSchemaProps<TCustomMetadata>,
```

Comment:
I don't this the names here are good.

Our schema system does define JavaScript properties on nodes based on the schema, and those properties are defined based on the `fields` parameter before this one. Adding a parameter called "props" could make someone thing this is specifies what properties should exist, and `NodeSchemaProps` sounds like it has something to do with nodes, and this really has nothing to do with nodes since it has no impact on node instance at all: it only impacts the schema.

Additionally, the name "props" is commonly used in frameworks like React for a specific purpose, and users of our library with such tools likely will have a function to convert nodes of a given schema into a props object: this sounds like its related to that, but is not.

If we are not trying to align with react "props", then we probably should avoid the term "props". As noted above I also think "properties" is misleading since this adds schema custom metadata, not properties on nodes.

Additionally this parameter is missing documentation.

While we could change the method name in the future, the TypeName `NodeSchemaProps` is something we will be stuck with. I think we should pick a better name.

Also, if we ever want to use some of the non-custom parts of the metadata in a strongly typed way (ex: use metadata to control API options), this API design won't work since it only captures strong types for the custom data.

Maybe for now it would make sense to go with something like:
`metadata?: SchemaMetadata<TCustomMetadata>,`

Then if we need to add strongly types non-custom member for specific schema, we can do something like:

`metadata?: ObjectSchemaMetadata<TCustomMetadata, TApiVersion, TWhatever>,`

The other option (Which avoids lots of generic parameters as more options are added) is just capturing the passed in type, and updating the extends clause to something like `TMetadata extends SchemaMetadata`. Then we retain all type information about the parameter passed in, and can filter out `TMetadata["custom"]` (or any other field) later if needed. This would be my preferred approach from a implementation perspective, though it might permit extra/unrecognized parameters on the non custom part, which we could make a runtime error but that might be a reason to prefer the the other approach.